automated-testing/e2e-testing/readme.md

```diff
@@ -0,0 +1,174 @@
+
+### Conditions
+
+Following activities should be performed as a part of building conditions based on user functions:
+
+- For each and every user functions, a set of conditions should be prepared.
+- Timing, data conditions and other factors that affect user functions can be considered as parameters.
+
+### Test Cases
+
+Following factors should be considered for building test cases:
+
+- For every scenario, one or more test cases should be created to test each and every functionality of the user functions. If possible, these test cases should be automated thru the standard CI/CD build pipeline processes with the track of each successful and failed build in AzDO.
+- Every single condition should be enlisted as a separate test case.
+
+## Applying the E2E testing [The How]
+
+Like any other testing, E2E testing also goes through formal planning, test execution, and closure phases.
+
+E2E testing is done with the following steps:
+
+### Planning
+
+- Business and Functional Requirement analysis
+- Test plan development
+- Test case development
+- Production like Environment setup for the testing
+- Test data setup
+- Decide exit criteria
+- Choose the testing methods that most applicable to your system. For the definiton of the various testing methods, please see [Testing Methods](./testing-methods.md) document.
+
+### Pre-requisite
+
+- System Testing should be complete for all the participating systems.
+- All subsystems should be combined together to work as a complete application.
+- Production like test environment should be ready.
+
+### Test Execution
+
+- Execute the test cases
+- Register the test results and decide on pass and failure
+- Report the Bugs in the bug reporting tool
+- Reverify the bug fixes
+
+### Test closure
+
+- Test report preparation
+- Evaluation of exit criteria
+- Test phase closure
+
+### Test Metrics
+
+The tracing the quality metrics gives insight about the current status of testing. Some of the common metrics of E2E testing are:
+
+- **Test case preparation status**: Number of test cases ready versus the total number of test cases.
+- **Frequent Test progress**: Number of test cases executed in the consistent frequent manner, e.g. weekly, versus a target number of the test cases in the same time period.
+- **Defects Status**: This metric represents the status of the defects found during testing. Defects should be logged into defect tracking tool (e.g. AzDO backlog) and resolved as per their severity and priority. Therefore, the percentage of open and closed defects as per their severity and priority should be calculated to track this metric. The AzDO Dashboard Query can be used to track this metric.
+- **Test environment availability**: This metric tracks the duration of the test environment used for end to end testing versus its scheduled allocation duration.
+
+## E2E Testing Frameworks and Tools
+
+### **1) Gauge Framework**
+
+![Gauge Framework](/images/gauge.jpg)
```

Comment:
All of the imges rom this point forward appear broken in preview. Images above look just fine. I'm thinking these may be missing the preceding "." in the path. Such as "./images/gauge.jpg" in stead of "/images/gauge.jpg".

Reply:
good catch. I've fixed all image links.